### ISSUE TITLE
Incomplete Coordinator's spec with Intel's SGX plugin

### ISSUE THREAD

**Issue by gianlu33** [edgelesssys/marblerun#179]
Hello everyone,

**Describe the bug**

I tried to install Marblerun on a local Minikube cluster with Intel's SGX plugin, but the Coordinator is not able to generate a quote and thus the attestation fails.

Logs:
```
[meshentry] invoking premain
[meshentry] invoking main
{"level":"info","ts":1623160373.8714085,"caller":"coordinator/run.go:46","msg":"starting coordinator","version":"0.3.2","commit":"2afe8c096e4d7beab060983ef7ebcd1ad2445c3d"}
{"level":"info","ts":1623160373.8714085,"caller":"coordinator/run.go:56","msg":"creating the Core object"}
{"level":"info","ts":1623160373.8714085,"caller":"core/core.go:122","msg":"loading state"}
{"level":"info","ts":1623160373.8714085,"caller":"core/core.go:139","msg":"No sealed state found. Proceeding with new state."}
{"level":"info","ts":1623160373.8834083,"caller":"core/core.go:360","msg":"generating quote"}
[error_driver2api sgx_enclave_common.cpp:249] Enclave not authorized to run, .e.g. provisioning enclave hosted in app without access rights to /dev/sgx_provision. You need add the user id to group sgx_prv or run the app as root.
[load_pce ../pce_wrapper.cpp:175] Error, call sgx_create_enclave for PCE fail [load_pce], SGXError:4004.
{"level":"warn","ts":1623160373.8994083,"caller":"core/core.go:363","msg":"Failed to get quote. Proceeding in simulation mode."}
```

**To reproduce**
Just install Marblerun in a cluster configured with Intel's SGX plugin

**Expected behavior**
Since the Coordinator uses "in-proc" attestation to generate the quote, the container needs special privileges to launch enclaves with the provision bit set. Using the Intel's plugin, this can be achieved by specifing the following annotation in the metadata of the Pod spec:

```yaml
    metadata:
      annotations:
        sgx.intel.com/quote-provider: <name_of_container>
```
The result of this annotation is that the operator injects in the container's limits this line: `sgx.intel.com/provision: 1`
You can also see [in the Intel's official repository](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/kustomization.yaml) that the above annotation is used for this purpose.

This is *not* happening by default in the latest version of Marblerun. I tried to install it both using `marblerun install` and helm.

As a consequence of this, the Coordinator is not able to generate the quote. Adding the annotation manually using `kubectl edit -n marblerun deployments.apps marblerun-coordinator` solves the issue, but of course this is a temporary solution.

**Environment:**
 - Marblerun version: 0.3.2
 
I am running Marblerun in a local Minikube cluster configured with Intel's SGX plugin.

**Additional context**
I don't know if this bug applies to the Marbles as well, but in general this annotation should be included in all the containers that run enclaves that perform in-process attestation.

**Comment by daniel-weisse** [comment]
Hi @gianlu33,

is your current user part of the `sgx_prv` group?
If not can you try adding the user: `sudo usermod -a -G sgx_prv <your_user_name>` and report back if that fixes the issue?

**Comment by gianlu33** [comment]
Hi,

Yes, my user is part of the `sgx_prv` group, and I have no problems running enclaves with the provision bit set from my user in the host OS. 

I guess this works differently in a containerized environment. This is why such annotation is used, i.e., to let the SGX plugin give a container the permissions to use the provision bit. But this is just my guess, so I may be mistaken.

**Comment by daniel-weisse** [comment]
You are right, when using the intel plugin the provision bit needs to be set.

I have [created a fix](https://github.com/edgelesssys/marblerun/tree/fix/cli-install-provision) which sets `sgx.intel.com/provision` for the coordinator, in case this plugin is used.

If you want to test, build the CLI from that branch, or simply run `go run cli/main.go install`.
Let me know if this works for you and we can merge this.


Regarding Marbles that use in-proc attestation: 
Since Marbles can use either in-process or out-of-process attestation, I think it should be left to the users to add the needed annotations/resources.
While we can inject resources/annotations into pods using our webhook, we can't differentiate between in-proc or out-of-proc from just the webhooks perspective. This means we would need to add annotations to let the webhook know it should inject these resources, which would be redundant since the user could instead directly add the resource/annotations to request in-proc or out-of-proc.

**Comment by gianlu33** [comment]
I can confirm that the fix solves the issue. Tried `go run cli/main.go install` and now the Coordinator can generate the quote successfully! About the Marbles you are right, it makes complete sense.

Btw, I also noticed just now that there is a misconfiguration of the Coordinator's EPC resource limits with Intel's plugin. The problem is that Intel's plugin defines the EPC memory in bytes, unlike with Azure's where the EPC memory is expressed in MiB. So, with Intel's plugin, the Coordinator is actually reserving just 10 bytes of EPC ;)
